addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -491,7 +497,7 @@ const Link = Widget.extend({
      */
     _updateLinkContent($link, linkInfos, { force = false } = {}) {
         if (force || (this._setLinkContent && (linkInfos.content !== this.data.originalText || linkInfos.url !== this.data.url))) {
-            if (linkInfos.content === this.data.originalText) {
+            if (linkInfos.content === this.data.originalText || (linkInfos.label === "" && this.data.isImage)) {
```

Comment:
We discussed with @lebl-odoo and we thought that it did not make sense to let the user write a label over an image, as the label will still overwrite the image. Therefore, I suggest to use `isImage` to completely hide the label field from the UI in that case so that it cannot replace the image at all. When that is done, you will be able to remove the check on `label` here and only check `isImage`.

Reply:
Noted :+1: